lib/arduino_ci/cpp_library.rb

```diff
@@ -37,11 +37,14 @@ class CppLibrary
 
     # @param base_dir [Pathname] The path to the library being tested
     # @param arduino_lib_dir [Pathname] The path to the libraries directory
-    def initialize(base_dir, arduino_lib_dir)
+    def initialize(base_dir, arduino_lib_dir, exclude_dirs = [])
       raise ArgumentError, 'base_dir is not a Pathname' unless base_dir.is_a? Pathname
       raise ArgumentError, 'arduino_lib_dir is not a Pathname' unless arduino_lib_dir.is_a? Pathname
+      raise ArgumentError, 'exclude_dir is not an array of Pathnames' unless exclude_dirs.is_a?(Array) &&
+                                                                             exclude_dirs.each { |p| p.is_a? Pathname }
```

Comment:
This should be [`.all?`](https://ruby-doc.org/core-2.4.1/Enumerable.html#method-i-all-3F), not [`.each`](https://ruby-doc.org/core-2.4.1/Array.html#method-i-each).